google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerDialectTableExporter.java

```diff
@@ -49,84 +49,78 @@
    *
    * @param spannerDialect a Cloud Spanner dialect.
    */
-  public SpannerTableExporter(SpannerDialect spannerDialect) {
+  public SpannerDialectTableExporter(SpannerDialect spannerDialect) {
     this.spannerDialect = spannerDialect;
     this.createTableTemplate =
         this.spannerDialect.getCreateTableString() + " {0} ({1}) PRIMARY KEY ({2})";
   }
 
   @Override
-  public String[] getSqlCreateStrings(Table table, Metadata metadata) {
-    /* The current implementation does not support UNIQUE constraints/indexes for relationships */
+  public String[] getSqlCreateStrings(ExportableTable table, JdbcServices jdbcServices) {
 
-    Table containingTable = getContainingTableForCollection(metadata, table);
-
-    Iterable<Column> keyColumns;
+    Collection<PhysicalColumn> keyColumns;
 
     if (table.hasPrimaryKey()) {
       // a typical table that corresponds to an entity type
       keyColumns = table.getPrimaryKey().getColumns();
-    } else if (containingTable != null) {
-      // a table that is actually an element collection property
-      keyColumns = table::getColumnIterator;
+    } else if (table.getForeignKeys().size() > 0) {
+      // a table with no PK's but has FK's; often corresponds to element collection properties
+      keyColumns = table.getPhysicalColumns();
```

Comment:
In hibernate 6, the table exporter does not get access to the metadata object as before, so one cannot do similar operation as `getContainingTableForCollection(...)`.

As a substitute, I modified it such that if a table does not have a PK but has at least 1 FK, then all of the columns in that table will be used as a PK for that table in Spanner. Not sure if we want to take this approach or go another way.